syntax = "proto3";

option go_package = "/";
service Greeter {
  rpc Recommendation(RecRequest) returns (ReqResponse) {}
}

message RecRequest{
  string userId = 1;
  int32 movieId = 2;
  int32 count = 3;
}

message ReqResponse{
  repeated MovieResponse movies = 1;
}

message MovieResponse{
  int32 movie_id = 1;
  float predicted_rating = 2;
}